src/brogue/Combat.c

```diff
@@ -1265,28 +1266,46 @@ void combatMessage(char *theMsg, color *theColor) {
     encodeMessageColor(newMsg, 0, theColor);
     strcat(newMsg, theMsg);
 
-    if (strLenWithoutEscapes(combatText) + strLenWithoutEscapes(newMsg) + 3 > DCOLS) {
-        // the "3" is for the semicolon, space and period that get added to conjoined combat texts.
+    length = strlen(combatText);
+
+    // Buffer combat messages here just for timing
+    if (length + strlen(newMsg) > COLS * 2 - 2) {
```

Comment:
I will flush out the comments a bit then.  There are two aspects, and I'm not sure which you mean.

Firstly, this is a bounds check.  The `- 2` accounts for the newline that `snprintf` will include a few lines down, as well as the null terminator that `strlen` will not count.

Secondly, why is the buffering there at all?  I experimented with removing it, and found that it reversed the order of arrival of combat messages.  In a combat round the player is accustomed to seeing their own action first, followed by the actions of monsters and other things.  For whatever reason, removing the logical delay implied by this buffer write (and eventual flush) seems to reverse that, which I thought would be jarring.  Rather than untangle the logic I opted to leave it alone.

Reply:
Thanks, I understand the buffering, I just think the comment is backwards, i.e. the comment says we are buffering messages but it's above a block which is emptying the buffer in case it over flows? It just seems a confusing place to annotate this. But it's not a problem, the function comment sets the stage for buffering now so it's clearer now :+1: